File: backend/src/main/java/com/jayway/kdag/graphql/DataStore.java
package com.jayway.kdag.graphql;

import com.couchbase.lite.*;

import java.io.IOException;
import java.util.HashMap;
import java.util.Map;

/**
 * Created by erik on 2017-04-21.
 */
public class DataStore {
    Manager manager = new Manager(new JavaContext(), Manager.DEFAULT_OPTIONS);
    Database database = manager.getDatabase("people");

    public DataStore() throws IOException, CouchbaseLiteException {
    }

    public void setup() throws CouchbaseLiteException, IOException {

        // Create a new document (i.e. a record) in the database.
        Document document = database.createDocument();
        Map properties = new HashMap();
        properties.put("firstName", "John");
        document.putProperties(properties);


        // Update a document.
        document.update(new Document.DocumentUpdater() {
            @Override
            public boolean update(UnsavedRevision newRevision) {
                Map properties = newRevision.getUserProperties();
                properties.put("firstName", "Johnny");
                newRevision.setUserProperties(properties);
                return true;
            }
        });

        // Delete a document.
        document.delete();

    }

    Database getDatabase() {
        return database;
    }
}

File: backend/src/main/java/com/jayway/kdag/graphql/DataStore.kt
package com.jayway.kdag.graphql

import com.couchbase.lite.*

import java.io.IOException
import java.util.HashMap

/**
 * Created by erik on 2017-04-21.
 */
class DataStore @Throws(IOException::class, CouchbaseLiteException::class)
constructor() {
    internal var manager = Manager(JavaContext(), Manager.DEFAULT_OPTIONS)
    internal var database = manager.getDatabase("people")

    @Throws(CouchbaseLiteException::class, IOException::class)
    fun setup() {

        // Create a new document (i.e. a record) in the database.
        val document = database.createDocument()
        val properties = mapOf("firstName" to "John")
        document.putProperties(properties)


        // Update a document.
        document.update { newRevision ->
            val properties = newRevision.userProperties
            properties["firstName"] = "Johnny"
            newRevision.userProperties = properties
            true
        }

        // Delete a document.
        document.delete()

    }
}
